client/components/search/index.jsx

```diff
@@ -293,42 +309,52 @@ const Search = React.createClass( {
 					}
 					aria-controls={ 'search-component-' + this.state.instanceId }
 					aria-label={ i18n.translate( 'Open Search', { context: 'button label' } ) }>
-				<Gridicon icon="search" className={ 'search-open__icon' + ( this.props.dir ? ' ' + this.props.dir : '' ) }/>
+				<Gridicon icon="search" className={ gridIconClass }/>
 				</div>
-				<input
-					type="search"
-					id={ 'search-component-' + this.state.instanceId }
-					className={ 'search__input' + ( this.props.dir ? ' ' + this.props.dir : '' ) }
-					placeholder={ placeholder }
-					role="search"
-					value={ searchValue }
-					ref="searchInput"
-					onChange={ this.onChange }
-					onKeyUp={ this.keyUp }
-					onKeyDown={ this.keyDown }
-					onFocus={ this.onFocus }
-					onBlur={ this.onBlur }
-					disabled={ this.props.disabled }
-					aria-hidden={ ! isOpenUnpinnedOrQueried }
-					autoCapitalize="none"
-					dir={ this.props.dir }
-					{ ...autocorrect } />
-				{ ( searchValue || this.state.isOpen ) ? this.closeButton() : null }
+				<div className={ fadeDivClass }>
```

Comment:
We want to fade text in input field. Currently CSS does not allow adding a pseudo class to replaced elements and input is one of them. So enclosing `div` was added and pseudo element `::before` added to this class.